tools/WindowStylesReportTool/WindowStylesReportTool.cpp

```diff
@@ -0,0 +1,414 @@
+    Logger::log(L"WS_EX_RTLREADING          {}", (exStyle & WS_EX_RTLREADING) == WS_EX_RTLREADING);
+    Logger::log(L"WS_EX_STATICEDGE          {}", (exStyle & WS_EX_STATICEDGE) == WS_EX_STATICEDGE);
+    Logger::log(L"WS_EX_TOOLWINDOW          {}", (exStyle & WS_EX_TOOLWINDOW) == WS_EX_TOOLWINDOW);
+    Logger::log(L"WS_EX_TOPMOST             {}", (exStyle & WS_EX_TOPMOST) == WS_EX_TOPMOST);
+    Logger::log(L"WS_EX_TRANSPARENT         {}", (exStyle & WS_EX_TRANSPARENT) == WS_EX_TRANSPARENT);
+    Logger::log(L"WS_EX_WINDOWEDGE          {}", (exStyle & WS_EX_WINDOWEDGE) == WS_EX_WINDOWEDGE);
+
+    Logger::log(L"");
+}
+
+void LogDwmAttributes(HWND window)
+{
+    Logger::log(L"------------------ DwmAttributes --------------------- ");
+    Logger::log(L"");
+
+    int intValue{};
+    unsigned int uintValue{};
+
+    LogDwmInfo(window, DWMWINDOWATTRIBUTE::DWMWA_NCRENDERING_ENABLED, intValue);
+    LogDwmInfo(window, DWMWINDOWATTRIBUTE::DWMWA_CLOAKED, intValue);
+
+    RECT rectValue{};
+    LogDwmRect(window, DWMWINDOWATTRIBUTE::DWMWA_CAPTION_BUTTON_BOUNDS, rectValue);
+    LogDwmRect(window, DWMWINDOWATTRIBUTE::DWMWA_EXTENDED_FRAME_BOUNDS, rectValue);
+
+    Logger::log(L"");
+}
+
+void LogVirtualDesktopInfo(HWND window)
+{
+    Logger::log(L"------------------ VirtualDesktop info --------------------- ");
+    Logger::log(L"");
+
+    IVirtualDesktopManager* vdManager = nullptr;
+    auto res = CoCreateInstance(CLSID_VirtualDesktopManager, nullptr, CLSCTX_ALL, IID_PPV_ARGS(&vdManager));
+    if (FAILED(res))
+    {
+        Logger::log(L"Failed to create VirtualDesktopManager instance");
+        Logger::log(L"");
+        return;
+    }
+
+    BOOL isWindowOnCurrentDesktop = false;
+    if (vdManager->IsWindowOnCurrentVirtualDesktop(window, &isWindowOnCurrentDesktop) == S_OK)
+    {
+        Logger::log(L"Window is on current virtual desktop: {}", isWindowOnCurrentDesktop);
+    }
+
+    GUID id{};
+    auto vdIdRes = vdManager->GetWindowDesktopId(window, &id);
+    if (vdIdRes == S_OK)
+    {
+        OLECHAR* guidString;
+        if (StringFromCLSID(id, &guidString) == S_OK)
+        {
+            Logger::log(L"Virtual desktop id: {}", guidString);
+        }
+
+        CoTaskMemFree(guidString);
+    }
+    else
+    {
+        Logger::log(L"GetWindowDesktopId error: {}", get_last_error_or_default(vdIdRes));
+    }
+
+    if (vdManager)
+    {
+        vdManager->Release();
+    }
+
+    Logger::log(L"");
+}
+
+void LogInfo(HWND window)
+{
+    auto processPath = get_process_path(window);
+    auto app = processPath;
+    auto pos = processPath.find_last_of('\\');
+    if (pos != std::string::npos && pos + 1 < processPath.length())
+    {
+        app = processPath.substr(pos + 1);
+    }
+
+    Logger::log(L"Window: {}", app);
+    Logger::log(L"");
+
+    LogStyles(window);
+    LogExStyles(window);
+    LogDwmAttributes(window);
+    LogVirtualDesktopInfo(window);
+
+    Logger::log(L"---------------------------------------");
```

Comment:
```suggestion
    Logger::log(L"=======================================");
```

This makes it easier to see when the window is finished and the next one starts in the log file.